daemon/emer-daemon.c

```diff
@@ -1547,6 +1580,29 @@ emer_daemon_new (const gchar             *persistent_cache_directory,
                        NULL);
 }
 
+gboolean
+emer_daemon_reset_tracking_id (EmerDaemon  *self,
+                               GError     **error)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+
+  if (priv->override_machine_id_path != NULL)
```

Comment:
Given the point of `EmerMachineIdProvider` is to abstract away how the machine ID is stored, I think it should have an `emer_machine_id_provider_reset_tracking_id` function. It already knows what the override path is, and the call to `emer_machine_id_provider_reload` to tell it the world has changed underneath its feet suggests that it should take care of the reset, too. This would also make the test case simpler because you don't need to hit the disk at all, and you can just assert that the mock version of this function was called.

Reply:
This is certainly a tension I was feeling too when implementing this. `EmerMachineIdProvider` takes a strv of potential override paths in priority order, so it can't really know which path to write to.

That is unless what we want to do is only have a single override path, which could be a smart move.